### PR TITLE
SEC-3.1: Authentication : Adding hostname configuration and encoding …

### PR DESCRIPTION
**Pull request by avaneesh90**
SEC-3.1: Authentication :
Adding hostname configuration before doing gnmi.get of hostname, not assuming that is pre configured

### CODE REVIEW

**Comment by greg-dennis** [comment]
Please rebase this PR forward to pickup the latests changes to the test, including the removal of the ssh_ip flag.

**Comment by greg-dennis** [comment]
I did a rebase myself and I am get `unexpected error for user "alice": rpc error: code = Unavailable desc = connection error: desc = "transport: Error while dialing: dial tcp 192.168.8.58:6030: connect: no route to host"` for every gNMI call that uses the modified context.

**Comment by avaneesh90** [comment]
> I did a rebase myself and I am get `unexpected error for user "alice": rpc error: code = Unavailable desc = connection error: desc = "transport: Error while dialing: dial tcp 192.168.8.58:6030: connect: no route to host"` for every gNMI call that uses the modified context.



> Please rebase this PR forward to pickup the latests changes to the test, including the removal of the ssh_ip flag.

Rebase is done and also updated the code changes to configure hostname alone, gnmi.get Encoding: gpb.Encoding_JSON_IETF, is already part of latest production script

**Comment by avaneesh90** [comment]
> I did a rebase myself and I am get `unexpected error for user "alice": rpc error: code = Unavailable desc = connection error: desc = "transport: Error while dialing: dial tcp 192.168.8.58:6030: connect: no route to host"` for every gNMI call that uses the modified context.

i am not seeing this issue in my environment.

File: feature/experimental/security/aaa/kne_tests/tls_authentication_over_grpc_test/tls_authentication_over_grpc_test.go [hunk 1]
```diff
@@ -134,6 +134,26 @@ func TestAuthentication(t *testing.T) {
 			if err != nil {
 				t.Fatal(err)
 			}
+
```
Comment (greg-dennis) [inline feature/experimental/security/aaa/kne_tests/tls_authentication_over_grpc_test/tls_authentication_over_grpc_test.go:137]: gofmt is complaining about the whitespace on this line

Comment (avaneesh90) [reply feature/experimental/security/aaa/kne_tests/tls_authentication_over_grpc_test/tls_authentication_over_grpc_test.go]: > gofmt is complaining about the whitespace on this line

Removed the whitespace

**Comment by greg-dennis** [comment]
I'm getting errors on all gNMI requests:
```
gnmi.Set unexpected error for user "alice": rpc error: code = Unavailable desc = error on request {
        replace:  {
          path:  {
            elem:  {
              name:  "system"
            }
            elem:  {
              name:  "config"
            }
            elem:  {
              name:  "hostname"
            }
          }
          val:  {
            json_ietf_val:  "\"ondatraDUT\""
          }
        }
        }: JGrpcServer: Error parsing metadata for RPC /gnmi.gNMI/Set
        ```

**Comment by greg-dennis** [comment]
@avaneesh90, what cptx image ID are you using?

**Review by greg-dennis — APPROVED** [review]

**Pull request merged**